### DISCUSSION TITLE
Query Regarding Performance Metrics and OLAP Workload Capability with Cassandra JDBC Driver

### DISCUSSION THREAD

**Discussion by Arjunguptha98** [ing-bank/cassandra-jdbc-wrapper#48] [Q&A]
Hello team,
I am exploring the use of the Cassandra JDBC driver for a project and have a few questions regarding its performance and capabilities, particularly in relation to OLAP workloads:

1.Performance Tests: Has there been any formal performance testing conducted using the Cassandra JDBC driver? If so, could you share some insights or metrics, specifically in terms of transactions per second (TPS) that were achieved during these tests?

2.OLAP Workload Support: How well does Cassandra, in conjunction with the JDBC driver, support OLAP workloads? I am interested in understanding the impact, if any, of running complex aggregation queries on the system. Does it handle such queries efficiently without significantly affecting the performance of OLAP workloads?

**Comment by maximevw** [comment]
Hello,

1. There is no specific performance tests for this driver. You have to know that this JDBC driver is just a wrapper of the [native Java driver for Cassandra](https://github.com/apache/cassandra-java-driver), so, the overhead induced by the JDBC driver is negligible compared to the performance of the native driver.
More information about performance can be found here: https://docs.datastax.com/en/developer/java-driver/4.17/manual/core/performance/
But, the topology of your database, the way you define your tables and querying them, the hardware used, ... may significantly vary the performance of your application. 
Some useful resources:
* https://cassandra.apache.org/doc/stable/cassandra/operating/hardware.html
* https://www.datastax.com/fr/blog/apache-cassandra-benchmarking-40-brings-heat-new-garbage-collectors-zgc-and-shenandoah

2. Cassandra can be very fast for reading/writing operations, but is not naturally optimized for OLAP, at least not standalone. By design, Cassandra is more a row-based, OLTP-oriented database.
You can read interesting answers here:
* https://stackoverflow.com/questions/48156537/using-cassandra-for-olap
* https://stackoverflow.com/questions/37434016/is-cassandra-for-olap-or-oltp-or-both
* https://dzone.com/articles/moving-data-from-cassandra-oltp-to-data-warehousin
